src/lib/zf/zf_stats.c

```diff
@@ -0,0 +1,69 @@
+/* SPDX-License-Identifier: MIT */
+/* SPDX-FileCopyrightText: (c) Advanced Micro Devices, Inc. */
+#include <zf_internal/zf_stack.h>
+
+/* ensure ef_vi/zf structures are the same size in case
+ * of adding/removing elements
+ */
+static_assert( sizeof(ef_vi_stats_layout) == sizeof(zf_stats_layout),
+               "ef_vi_stats_layout and zf_stats_layout size mismatch");
+
+static_assert( offsetof(zf_stats_field_layout, evsfl_name) == offsetof(ef_vi_stats_field_layout, evsfl_name),
+               "zf_stats_field_layout member evsfl_name does not match ef_vi_stats_field_layout");
+static_assert( offsetof(zf_stats_field_layout, evsfl_offset) == offsetof(ef_vi_stats_field_layout, evsfl_offset),
+               "zf_stats_field_layout member evsfl_name does not match ef_vi_stats_field_layout");
+static_assert( offsetof(zf_stats_field_layout, evsfl_size) == offsetof(ef_vi_stats_field_layout, evsfl_size),
+               "zf_stats_field_layout member evsfl_name does not match ef_vi_stats_field_layout");
+
+static_assert( offsetof(zf_stats_layout, evsl_data_size) == offsetof(ef_vi_stats_layout, evsl_data_size),
+               "zf_stats_field_layout member evsfl_name does not match ef_vi_stats_field_layout");
+static_assert( offsetof(zf_stats_layout, evsl_fields_num) == offsetof(ef_vi_stats_layout, evsl_fields_num),
+               "zf_stats_field_layout member evsfl_name does not match ef_vi_stats_field_layout");
+static_assert( offsetof(zf_stats_layout, evsl_fields) == offsetof(ef_vi_stats_layout, evsl_fields),
+               "zf_stats_field_layout member evsfl_name does not match ef_vi_stats_field_layout");
+
+
+int
+zf_stats_query_layout(struct zf_stack* stack,
+                      const zf_stats_layout** const layout_out, int layout_sz)
```

Comment:
Rather than have the user get interface count and malloc, I would consider making this zf_alloc_stats_query_layout() and return a new data type with layout(s) and size.  Would need a corresponding zf_free_stats_query_layout()

Reply:
My personal preference is to have a specific free call to make it symmetric. Makes code easier to follow (IMO) and in theory the data structure could contain pointers to other objects that needed to be freed first in future. (I'd also advise taking a look in the library for any precedents one way or the other).